blueocean-dashboard/src/main/js/components/Pipelines.jsx

```diff
@@ -18,7 +18,12 @@ export default class Pipelines extends Component {
         }
 
         const pipelineRecords = pipelines
-            .map(data => new PipelineRecord(data))
+            .map(data =>
+                'numberOfFolders' in data ?
+                    new FolderRecord(data) :
+                    new PipelineRecord(data)
+            )
+            .filter(data => !(data instanceof FolderRecord))
```

Comment:
Why not just `.filter(data => numberOfFolders in data)` instead of all this shenanigans creating extra stuff that's thrown away?